Modify compilation script to add new variable path to LDFLAGS

## Describe the Task ##
Testing the installation of MET on WCOSS2 Acorn using Python 3.10.4 would not work without the addition of a new environment variable.

Running:
```
module load intel python/3.10.4 
module load intel
module load ve/evs/1.0
/apps/spack/python/3.10.4/intel/19.1.3.304/xqft4d45h4dp4xnbz2ue3nbxv65i6bgp/bin/python3-config --ldflags
```
on Acorn gave the following output:
```
-L/apps/spack/python/3.10.4/intel/19.1.3.304/xqft4d45h4dp4xnbz2ue3nbxv65i6bgp/lib64  -lcrypt -lintl -ldl  -lutil -lm -lm
```
However, when adding this information to MET_PYTHON_LD (along with an additional path and the python library):
```
export MET_PYTHON_LD=-L/apps/spack/python/3.10.4/intel/19.1.3.304/xqft4d45h4dp4xnbz2ue3nbxv65i6bgp/lib/python3.10/config-3.10-x86_64-linux-gnu/ -L/apps/spack/python/3.10.4/intel/19.1.3.304/xqft4d45h4dp4xnbz2ue3nbxv65i6bgp/lib64 -lpython3.10 -lintl -lcrypt -ldl -lutil -lm -lm
```
led to the following error:
```
/usr/lib64/gcc/x86_64-suse-linux/7/../../../../x86_64-suse-linux/bin/ld: cannot find -lintl: No such file or directory
```
Running _ldd /apps/prod/ve/intel/19.1.3.304/python/3.10.4/evs/1.0/bin/python_ resulted in locating this path:
```
libintl.so.8 => /apps/spack/gettext/0.21/intel/19.1.3.304/at2kdo4edvuhyzrt5g6zhwrdb7bdui4s/lib64/libintl.so.8 (0x0000150434aef000)
```
And adding a new environment variable to the configuration file:
```
export INTL_DIR=/apps/spack/gettext/0.21/intel/19.1.3.304/at2kdo4edvuhyzrt5g6zhwrdb7bdui4s/lib64
```
and modifying the value of LDFLAGS to include this path in the compile_MET_all.sh script resolved the problem and resulted in a successful compilation.

### Time Estimate ###
<30 min

### Sub-Issues ###
Consider breaking the task down into sub-issues.
- [ ] *Add a checkbox for each sub-issue here.*

### Relevant Deadlines ###
MET-11.1.0-beta1

### Funding Source ###
Split 2792542, 2773542

## Define the Metadata ##

### Assignee ###
- [x] Select **engineer(s)** or **no engineer** required
- [ ] Select **scientist(s)** or **no scientist** required

### Labels ###
- [x] Select **component(s)**
- [x] Select **priority**
- [x] Select **requestor(s)**

### Projects and Milestone ###
- [x] Select **Repository** and/or **Organization** level **Project(s)** or add **alert: NEED PROJECT ASSIGNMENT** label
- [x] Select **Milestone** as the next official version or **Future Versions**

## Define Related Issue(s) ##
Consider the impact to the other METplus components.
- [x] [METplus](https://github.com/dtcenter/METplus/issues/new/choose), [MET](https://github.com/dtcenter/MET/issues/new/choose), [METdataio](https://github.com/dtcenter/METdataio/issues/new/choose), [METviewer](https://github.com/dtcenter/METviewer/issues/new/choose), [METexpress](https://github.com/dtcenter/METexpress/issues/new/choose), [METcalcpy](https://github.com/dtcenter/METcalcpy/issues/new/choose), [METplotpy](https://github.com/dtcenter/METplotpy/issues/new/choose)

## Task Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [ ] Complete the issue definition above, including the **Time Estimate** and **Funding Source**.
- [ ] Fork this repository or create a branch of **develop**.
Branch name: `feature_<Issue Number>_<Description>`
- [ ] Complete the development and test your changes.
- [ ] Add/update log messages for easier debugging.
- [ ] Add/update unit tests.
- [ ] Add/update documentation.
- [ ] Push local changes to GitHub.
- [ ] Submit a pull request to merge into **develop**.
Pull request: `feature <Issue Number> <Description>`
- [ ] Define the pull request metadata, as permissions allow.
Select: **Reviewer(s)** and **Development** issues
Select: **Repository** level development cycle **Project** for the next official release
Select: **Milestone** as the next official version
- [ ] Iterate until the reviewer(s) accept and merge your changes.
- [ ] Delete your fork or branch.
- [ ] Close this issue.

--- a/internal/scripts/installation/compile_MET_all.sh
+++ b/internal/scripts/installation/compile_MET_all.sh
@@ -832,9 +832,9 @@ if [ $COMPILE_MET -eq 1 ]; then
   # https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html
   # ${parameter:+word}
   # If parameter is null or unset, nothing is substituted, otherwise the expansion of word is substituted.
-  export LDFLAGS="${LDFLAGS} -Wl,-rpath,${LIB_DIR}/lib${MET_NETCDF:+:$MET_NETCDF/lib}${MET_HDF5:+:$MET_HDF5/lib}${MET_BUFRLIB:+:$MET_BUFRLIB}${MET_GRIB2CLIB:+:$MET_GRIB2CLIB}${MET_PYTHON:+:$MET_PYTHON/lib}${MET_GSL:+:$MET_GSL/lib}"
+  export LDFLAGS="${LDFLAGS} -Wl,-rpath,${LIB_DIR}/lib${ADDTL_DIR:+:$ADDTL_DIR}${MET_NETCDF:+:$MET_NETCDF/lib}${MET_HDF5:+:$MET_HDF5/lib}${MET_BUFRLIB:+:$MET_BUFRLIB}${MET_GRIB2CLIB:+:$MET_GRIB2CLIB}${MET_PYTHON:+:$MET_PYTHON/lib}${MET_GSL:+:$MET_GSL/lib}"
   export LDFLAGS="${LDFLAGS} -Wl,-rpath,${LIB_JASPER:+$LIB_JASPER}${LIB_LIBPNG:+:$LIB_PNG}${LIB_Z:+$LIB_Z}"
-  export LDFLAGS="${LDFLAGS} ${LIB_JASPER:+-L$LIB_JASPER} ${LIB_LIBPNG:+-L$LIB_LIBPNG} ${MET_HDF5:+-L$MET_HDF5/lib}"
+  export LDFLAGS="${LDFLAGS} ${LIB_JASPER:+-L$LIB_JASPER} ${LIB_LIBPNG:+-L$LIB_LIBPNG} ${MET_HDF5:+-L$MET_HDF5/lib} ${ADDTL_DIR:+-L$ADDTL_DIR}"
   export LIBS="${LIBS} -lhdf5_hl -lhdf5 -lz"
   export MET_FONT_DIR=${TEST_BASE}/fonts
 

--- a/internal/scripts/installation/config/install_met_env.acorn_py3.10
+++ b/internal/scripts/installation/config/install_met_env.acorn_py3.10
@@ -1,0 +1,36 @@
+module use /apps/ops/para/libs/modulefiles/compiler/intel/19.1.3.304/
+module load intel python/3.10.4
+module load ve/evs/1.0
+module load netcdf/4.7.4
+module load hdf5/1.12.2
+module load bufr/11.5.0
+module load zlib/1.2.11
+module load jasper/2.0.25
+module load libpng/1.6.37
+module load gsl/2.7
+module load g2c/1.6.4
+
+export TEST_BASE=/apps/sw_review/emc/MET/11.0.1
+export LIB_DIR=${TEST_BASE}/external_libs
+export BIN_DIR_PATH=${TEST_BASE}/exec
+export COMPILER=intel_19.1.3.304
+export MET_SUBDIR=${TEST_BASE}
+export MET_TARBALL=v11.0.1.tar.gz
+export USE_MODULES=TRUE
+export ADDTL_DIR=/apps/spack/gettext/0.21/intel/19.1.3.304/at2kdo4edvuhyzrt5g6zhwrdb7bdui4s/lib64
+export PYTHON_MODULE=python_3.10.4
+export MET_PYTHON=/apps/spack/python/3.10.4/intel/19.1.3.304/xqft4d45h4dp4xnbz2ue3nbxv65i6bgp
+export MET_PYTHON_CC=-I/apps/spack/python/3.10.4/intel/19.1.3.304/xqft4d45h4dp4xnbz2ue3nbxv65i6bgp/include/python3.10
+export MET_PYTHON_LD=-L/apps/spack/python/3.10.4/intel/19.1.3.304/xqft4d45h4dp4xnbz2ue3nbxv65i6bgp/lib/python3.10/config-3.10-x86_64-linux-gnu/\ -L/apps/spack/python/3.10.4/intel/19.1.3.304/xqft4d45h4dp4xnbz2ue3nbxv65i6bgp/lib64\ -lpython3.10\ -lintl\ -lcrypt\ -ldl\ -lutil\ -lm\ -lm
+export MET_NETCDF=/apps/prod/hpc-stack/intel-19.1.3.304/netcdf/4.7.4
+export MET_HDF5=/apps/prod/hpc-stack/intel-19.1.3.304/hdf5/1.12.2
+export MET_BUFRLIB=/apps/ops/prod/libs/intel/19.1.3.304/bufr/11.5.0/lib64
+export MET_GRIB2CLIB=/apps/ops/prod/libs/intel/19.1.3.304/g2c/1.6.4/lib64
+export MET_GRIB2CINC=/apps/ops/prod/libs/intel/19.1.3.304/g2c/1.6.4/include
+export MET_GSL=/apps/spack/gsl/2.7/intel/19.1.3.304/xks7dxbowrdxhjck5zxc4rompopocevb
+export BUFRLIB_NAME=-lbufr_4
+export GRIB2CLIB_NAME=-lg2c
+export LIB_JASPER=/apps/spack/jasper/2.0.25/intel/19.1.3.304/sjib74krrorkyczqpqah4tvewmlnqdx4/lib64
+export LIB_LIBPNG=/apps/spack/libpng/1.6.37/intel/19.1.3.304/4ohkronuhlyherusoszzrmur5ewvlwzh/lib
+export LIB_Z=/apps/spack/zlib/1.2.11/intel/19.1.3.304/hjotqkckeoyt6j6tibalwzrlfljcjtdh/lib
+export SET_D64BIT=FALSE

--- a/internal/scripts/installation/config/install_met_env.wcoss2_py3.10
+++ b/internal/scripts/installation/config/install_met_env.wcoss2_py3.10
@@ -1,0 +1,47 @@
+# JY module use /apps/ops/para/libs/modulefiles/compiler/intel/19.1.3.304/
+module load intel
+# JY add following two
+module load craype/2.7.13
+module load cray-mpich/8.1.12
+
+module load python/3.10.4
+module load netcdf/4.7.4
+module load hdf5/1.10.6
+module load bufr/11.6.0
+module load zlib/1.2.11
+module load jasper/2.0.25
+module load libpng/1.6.37
+module load gsl/2.7
+module load g2c/1.6.4
+
+#export TEST_BASE=/apps/ops/para/libs/intel/19.1.3.304/met/11.0.1
+export TEST_BASE=$(pwd)
+export LIB_DIR=${TEST_BASE}/external_libs
+export BIN_DIR_PATH=${TEST_BASE}/bin
+export COMPILER=intel_19.1.3.304
+export MET_SUBDIR=${TEST_BASE}
+export MET_TARBALL=v11.0.1.tar.gz
+export USE_MODULES=TRUE
+export ADDTL_DIR=/apps/spack/gettext/0.21/intel/19.1.3.304/at2kdo4edvuhyzrt5g6zhwrdb7bdui4s/lib64
+export PYTHON_MODULE=python_3.10.4
+export MET_PYTHON=/apps/spack/python/3.10.4/intel/19.1.3.304/xqft4d45h4dp4xnbz2ue3nbxv65i6bgp
+export MET_PYTHON_CC=-I/apps/spack/python/3.10.4/intel/19.1.3.304/xqft4d45h4dp4xnbz2ue3nbxv65i6bgp/include/python3.10
+export MET_PYTHON_LD=-L/apps/spack/python/3.10.4/intel/19.1.3.304/xqft4d45h4dp4xnbz2ue3nbxv65i6bgp/lib\ -lpython3.10\ -lpthread\ -ldl\ -lutil\ -lm\ -Xlinker\ -export-dynamic
+export MET_NETCDF=/apps/prod/hpc-stack/intel-19.1.3.304/netcdf/4.7.4
+# JY export MET_HDF5=/apps/prod/hpc-stack/intel-19.1.3.304/hdf5/1.12.2
+export MET_HDF5=${HDF5_ROOT}
+export MET_BUFRLIB=/apps/ops/prod/libs/intel/19.1.3.304/bufr/11.6.0/lib64
+# JY export MET_GRIB2CLIB=/apps/ops/prod/libs/intel/19.1.3.304/g2c/1.6.4/lib64
+# JY export MET_GRIB2CINC=/apps/ops/prod/libs/intel/19.1.3.304/g2c/1.6.4/include
+export MET_GRIB2CLIB=${g2c_ROOT}/lib64
+export MET_GRIB2CINC=${G2C_INC}
+export MET_GSL=/apps/spack/gsl/2.7/intel/19.1.3.304/xks7dxbowrdxhjck5zxc4rompopocevb
+export BUFRLIB_NAME=-lbufr_4
+export GRIB2CLIB_NAME=-lg2c
+# JY export LIB_JASPER=/apps/spack/jasper/2.0.25/intel/19.1.3.304/sjib74krrorkyczqpqah4tvewmlnqdx4/lib64
+export LIB_JASPER=${JASPER_LIBDIR}
+# JY export LIB_LIBPNG=/apps/spack/libpng/1.6.37/intel/19.1.3.304/4ohkronuhlyherusoszzrmur5ewvlwzh/lib
+export LIB_LIBPNG=${LIBPNG_LIBDIR}
+# JY export LIB_Z=/apps/spack/zlib/1.2.11/intel/19.1.3.304/hjotqkckeoyt6j6tibalwzrlfljcjtdh/lib
+export LIB_Z=${ZLIB_LIBDIR}
+export SET_D64BIT=FALSE

--- a/internal/scripts/installation/modulefiles/11.0.1.lua.wcoss2
+++ b/internal/scripts/installation/modulefiles/11.0.1.lua.wcoss2
@@ -1,0 +1,27 @@
+help([[
+]])
+
+local pkgName    = myModuleName()
+local pkgVersion = myModuleVersion()
+local pkgNameVer = myModuleFullName()
+
+local hierA        = hierarchyA(pkgNameVer,1)
+local compNameVer  = hierA[1]
+
+
+conflict(pkgName)
+
+local opt = os.getenv("HPC_OPT") or os.getenv("OPT") or "/opt/modules"
+
+local base = pathJoin(opt,compNameVer,pkgName,pkgVersion)
+
+prepend_path("PATH", pathJoin(base,"bin"))
+
+setenv("MET_ROOT", base)
+setenv("MET_BASE", pathJoin(base, "share", "met"))
+setenv("MET_VERSION", pkgVersion)
+
+whatis("Name: ".. pkgName)
+whatis("Version: " .. pkgVersion)
+whatis("Category: applications")
+whatis("Description: Model Evaluation Tools (MET)")

--- a/internal/scripts/installation/modulefiles/11.0.1_acorn
+++ b/internal/scripts/installation/modulefiles/11.0.1_acorn
@@ -1,0 +1,34 @@
+#%Module######################################################################
+##
+##      Model Evaluation Tools
+##
+proc ModulesHelp { } {
+  puts stderr "Sets up the paths and environment variables to use the Model Evaluation Tools v11.0.1
+     *** For help see the official MET webpage at http://www.dtcenter.org/met/users ***"
+}
+
+# The intel compiler is required to run MET
+
+module use /apps/ops/para/libs/modulefiles/compiler/intel/19.1.3.304/
+module load intel python/3.10.4
+module load ve/evs/1.0
+module load netcdf/4.7.4
+module load hdf5/1.12.2
+module load bufr/11.5.0
+module load zlib/1.2.11
+module load jasper/2.0.25
+module load libpng/1.6.37
+module load gsl/2.7
+module load g2c/1.6.4
+
+set base /apps/sw_review/emc/MET/11.0.1
+set ver  11.0.1
+set share $base/share/met
+set lib_base $base
+
+prepend-path PATH $base/exec
+
+setenv METversion V$ver
+setenv MET_ROOT $base
+
+

